test/jaxp/javax/xml/jaxp/unittest/validation/SchemaTest.java

```diff
@@ -31,21 +31,94 @@
 import javax.xml.transform.stream.StreamSource;
 import javax.xml.validation.SchemaFactory;
 import org.testng.Assert;
-import org.testng.annotations.Listeners;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 import org.xml.sax.ErrorHandler;
 import org.xml.sax.SAXException;
 import org.xml.sax.SAXParseException;
 
 /*
  * @test
- * @bug 8149915 8222991
+ * @bug 8149915 8222991 8282280 8144117
  * @library /javax/xml/jaxp/libs /javax/xml/jaxp/unittest
  * @run testng/othervm validation.SchemaTest
- * @summary Test Schema creation
+ * @summary Tests schemas and Schema creation.
  */
-@Listeners({jaxp.library.FilePolicy.class})
 public class SchemaTest {
+    private static final String XSD_8144117 = "<?xml version='1.0'?>\n" +
+        "<xs:schema xmlns:xs='http://www.w3.org/2001/XMLSchema'>\n" +
+        "\n" +
+        "  <xs:simpleType name='mjtype1'>\n" +
+        "    <xs:restriction base='xs:string'>\n" +
+        "      <xs:minLength value='3'/>\n" +
+        "    </xs:restriction>\n" +
+        "  </xs:simpleType>\n" +
+        "\n" +
+        "  <xs:simpleType name='mjtype2'>\n" +
+        "    <xs:restriction base='mjtype1'>\n" +
+        "      <xs:minLength value='3'/>\n" +
+        "      <xs:length value='4'/>\n" +
+        "    </xs:restriction>\n" +
+        "  </xs:simpleType>\n" +
+        "\n" +
+        "  <xs:element name='top' type='mjtype2'/>\n" +
+        "\n" +
+        "</xs:schema>";
+
+    /*
+     * DataProvider: valid xsd
+     */
+    @DataProvider(name = "xsd")
+    Object[][] getXSDString() {
+        return new Object[][]{
+            {XSD_8144117},
+        };
+    }
+
+    /*
+     * DataProvider: valid xsd
```

Comment:
This comment is the same as the above provider. Can we change it to a more meaningful one?

Reply:
Updated with a bit more description.